test/unit/plotly-dash-preview_test.js

```diff
@@ -0,0 +1,120 @@
+const tap = require('tap')
+const sinon = require('sinon')
+
+const _module = require('../../src/component/plotly-dash-preview')
+const remote = require('../../src/util/remote')
+const { createMockWindow } = require('../common')
+
+tap.test('parse:', t => {
+  const fn = _module.parse
+
+  t.test('should error when no *url* field is given', t => {
+    const shouldFail = ['', null, true, 1, {}]
+
+    shouldFail.forEach(d => {
+      t.test(`(case ${JSON.stringify(d)})`, t => {
+        fn({url: d}, {}, (errorCode, result) => {
+          t.equal(errorCode, 400, 'code')
+          t.end()
+        })
+      })
+    })
+
+    t.end()
+  })
+
+  t.test('should default to PNG when no format specified', t => {
+    fn({url: 'dummy'}, {}, (errorCode, result) => {
+      t.equal(errorCode, null, 'code')
+      t.equal(result.format, 'png')
+      t.end()
+    })
+  })
+
+  t.end()
+})
+
+tap.test('render:', t => {
+  const fn = _module.render
+  
+  t.afterEach((done) => {
+    remote.createBrowserWindow.restore()
+    done()
+  })
+
+  t.test('should call printToPDF', t => {
+    const win = createMockWindow()
+    sinon.stub(remote, 'createBrowserWindow').returns(win)
+    win.webContents.printToPDF.yields(null, '-> image data <-')
+
+    fn({
+      width: 500,
+      height: 500,
+      url: 'dummy',
+      format: 'pdf'
+    }, {}, (errorCode, result) => {
+      t.ok(win.webContents.printToPDF.calledOnce)
```

Comment:
This fails because `printToPDF` waits for `page-favicon-updated` event before being called. Is there a way to handle this? The next tests also fail for similar reason.. !!

Reply:
`win` is an event emitter as created [here](https://github.com/plotly/image-exporter/blob/934999da692e2cdc61fdf7debdb87e022330f640/test/common.js#L25-L43), so calling `win.emit('page-favicon-updated')` should allow you to test the callback.